inst/tests/tests.Rraw

```diff
@@ -11146,16 +11146,20 @@ DT = data.table(col1 = c(1,1,1, 2,2,2), col2 = c(2,2,2,1,1,1), ID = c(rep(1,3),
 test(1863.2, DT[, lapply(.SD, var), by=ID], data.table(ID=c(1,2), col1=0, col2=0, key="ID"))
 
 # Fix the bug when keys contain non UTF8 strings #2566 #2462 #1826
-# Only on Windows platform it might fail, because other platforms use UTF8 as the native encoding.
-DT <- fread(file = testDir("issue_2566.csv"), encoding = "UTF-8")
-# `fread` return a utf-8 encoded data, we should convert x to native encoding.
-# However, we need this condition to ensure the native encoding can be used for Chinese characters.
-# Otherwise, the test will fail because the strings have been damaged.
-if (identical(enc2utf8(enc2native(DT$x)), DT$x)) DT[, x:= enc2native(x)]
-setkey(DT, x)
-test(1864.1, DT[J("\u516c\u5141\u4ef7\u503c\u53d8\u52a8\u635f\u76ca"), z], 1L)
-setkey(DT, y)
-test(1864.2, DT[J("\u516c\u5141\u4ef7\u503c\u53d8\u52a8\u635f\u76ca"), z], 1L)
+utf8_strings <- c("\u00e7ile", "fa\u00e7ile", "El. pa\u00c5\u00a1tas", "\u00a1tas", "\u00de")
+latin1_strings <- iconv(utf8_strings, from = "UTF-8", to = "latin1")
+mixed_strings <- c(utf8_strings, latin1_strings)
+DT1 <- data.table(x = mixed_strings, y = c(latin1_strings, utf8_strings), z = 1:10)
+DT2 <- copy(DT1)
+setkey(DT1, x)
+setkey(DT2, y)
+test(1864.1, DT1$x, sort(c(utf8_strings, utf8_strings), method = "radix"))
+test(1864.2, DT2$y, sort(c(utf8_strings, utf8_strings), method = "radix"))
+test(1864.3, DT1[J(utf8_strings)], DT1[J(latin1_strings)])
```

Comment:
Imagine that, for reason of a strange bug, both `DT1[J(utf8_strings)]` and `DT1[J(latin1_strings)]` returned NULL.  The test would still pass. I like this pattern for that reason : 
test(1864.3,  DT1[J(utf8_strings)],  ans<- data.table( .. known answer .. ) )
test(1864.4,  DT1[J(latin1_strings)], ans)

Reply:
Perhaps use the integers in `z` to construct `ans`.

Reply:
Otherwise, great!

Reply:
You're right. Using an explicit answer is indeed better.